src/lib/utils/calendar.cpp

```diff
@@ -34,6 +34,39 @@ std::tm do_gmtime(std::time_t time_val)
    return tm;
    }
 
+#if !defined(BOTAN_TARGET_OS_IS_WINDOWS)
+time_t posix_timegm(struct tm *tm)
+   {
+   time_t out;
+   std::string tz_backup;
+
+   // Store current value of env variable TZ
+   auto tz_env_pointer = getenv("TZ");
+   if (tz_env_pointer)
+      tz_backup = std::string(tz_env_pointer);
+
+   // Clear value of TZ
+   setenv("TZ", "", 1);
+   tzset();
```

Comment:
I know this is what the man page recommends as the portable alternative. But changing the timezone globally across the process seems really dangerous to me, in a multithreaded context completely unrelated threads would briefly see the timezone change.

If the main reason for this is Android, I'd say skip this entirely and on that platform use `mktime` and accept being off by a few hours (or perhaps there is some other way to get the TZ offset easily in the bionic C library).